src/browser/browser.ts

```diff
@@ -230,9 +349,27 @@ export class Browser {
       this.log.debug('Taking screenshot', 'filePath', options.filePath);
     }
 
-    await this.withTimingMetrics<void>(() => {
-      return page.screenshot({ path: options.filePath });
-    }, 'screenshot');
+    const screenshotFn = () =>
+      page.screenshot({ path: options.filePath, fullPage: options.fullPageImage, captureBeyondViewport: options.fullPageImage });
+    await this.withTimingMetrics(
+      scrollResult.scrolled ? this.withFullPageViewport(screenshotFn, page, options, scrollResult.scrollHeight) : screenshotFn,
+      'screenshot'
+    );
```

Comment:
*This comment got lost when submitting the review...*

This is a bit hard to read and I don't think we need to set back the viewport to its original state in case of a full page screenshot. Could we simplify with something like this?
```suggestion
    await this.withTimingMetrics(async () => {
      if (scrollResult.scrolled) {
        await this.setViewport(page, {
          ...options,
          height: scrollResult.scrollHeight,
        });
      }

      return page.screenshot({ path: options.filePath, fullPage: options.fullPageImage, captureBeyondViewport: options.fullPageImage });
    }, 'screenshot');
```

Reply:
Removed. The viewport reset was there because I ported the code from the hackathon version, where we were also reusing the page to take screenshots of each panel